test/foundry-tests/Fees.t.sol

```diff
@@ -581,6 +594,47 @@ contract FeesTest is Test, Deployers, TokenFixture, GasSnapshot {
         assertEq(currency1.balanceOf(address(hook)), expectedHookFees);
     }
 
+    function testCollectFeesForNativeTokens() public {
+        uint16 protocolFee = _computeFee(_oneForZero, 10); // 10% on 0 to 1 swaps
+        protocolFeeController.setSwapFeeForPool(key0.toId(), protocolFee);
+        manager.setProtocolFeeController(IProtocolFeeController(protocolFeeController));
+        manager.setProtocolFees(key0);
+
+        (Pool.Slot0 memory slot0,,,) = manager.pools(key0.toId());
+        assertEq(getSwapFee(slot0.protocolFees), protocolFee);
+
+        uint16 hookFee = _computeFee(_oneForZero, 5); // 20% on 1 to 0 swaps
+        hook.setSwapFee(key0, hookFee);
+        manager.setHookFees(key0);
+
+        (slot0,,,) = manager.pools(key0.toId());
+        assertEq(getSwapFee(slot0.hookFees), hookFee);
+
+        IPoolManager.ModifyPositionParams memory params = IPoolManager.ModifyPositionParams(-120, 120, 10e18);
+        modifyPositionRouter.modifyPosition{value: 10e18}(key0, params, ZERO_BYTES);
+        // 1 for 0 
+        MockERC20(Currency.unwrap(currency1)).approve(address(swapRouter), type(uint256).max);
+        swapRouter.swap(
+            key0,
+            IPoolManager.SwapParams(false, 10000, TickMath.MAX_SQRT_RATIO - 1),
+            PoolSwapTest.TestSettings(true, true),
+            ZERO_BYTES
+        );
+
+        uint256 expectedProtocolFees = 3; // 10% of 30 is 3
+        vm.prank(address(protocolFeeController));
+        manager.collectProtocolFees(address(protocolFeeController), currency1, 0);
```

Comment:
Shouldn't we be collecting `expectedProtocolFees ` here and not 0? I'm surprised the assertion on the next line passes

Reply:
At [L#89](https://github.com/Uniswap/v4-core/blob/9b912322d1456b9b2a7abd2585fd430ce19a12c5/contracts/Fees.sol#L89) `contracts/Fees.sol`, if the `amount` is zero, the function collects all protocol fees accrued. So the next `assert` should not generate error.